samples/nrf9160/icalendar_parser/src/main.c

```diff
@@ -0,0 +1,644 @@
+#include <modem/modem_key_mgmt.h>
+#include <net/socket.h>
+#include <net/http_client.h>
+#include <net/tls_credentials.h>
+#include <net/icalendar_parser.h>
+#include <modem/modem_info.h>
+
+#define UI_BUTTON_1                     DK_BTN1_MSK
+#define UI_BUTTON_2                     DK_BTN2_MSK
+#define UI_SWITCH_1                     DK_BTN3_MSK
+#define UI_SWITCH_2                     DK_BTN4_MSK
+
+/* Interval to blink calendar state LED */
+#define ICAL_STATE_LED_ON_PERIOD                500
+#define ICAL_STATE_LED_OFF_PERIOD               500
+/* Interval to check current date-time information */
+#define ICAL_DATETIME_PERIOD                    1000
+/* Interval to check current calendar event and set LED */
+#define ICAL_EVENT_LED_PERIOD                   1000
+
+/* Downloading indicator */
+static bool downloading;
+
+/* Receiving buffer for HTTP client */
+#define MAX_RECV_BUF_LEN 1024
+static u8_t http_recv_buf[MAX_RECV_BUF_LEN];
+
+static struct icalendar_parser ical;
+static struct k_delayed_work ical_work;
+static struct k_delayed_work datetime_update_work;
+
+/* Current date-time information */
+static u32_t current_date;
+static u32_t current_time;
+static u8_t timezone;
+
+/** List of pointers to the allocated memory for parsed calendar components */
+static struct ical_component *ical_coms[CONFIG_MAX_PARSED_COMPONENTS];
+/** Total number of parsed calendar components */
+static size_t component_cnt;
+
+/* Certificate for calendar host */
+static const char cert[] = {
+	#include "../cert/certificates"
+};
+
+/**@brief Read current time from modem */
+static void datetime_work_handler(struct k_work *unused)
+{
+	int err;
+	char recv_buf[CONFIG_MODEM_INFO_BUFFER_SIZE] = { 0 };
+
+	err = modem_info_string_get(MODEM_INFO_DATE_TIME, recv_buf);
+	if (err <= 0) {
+		printk("read modem info error\n");
+	} else {
+		int day = 0, month = 0, year = 0, hour = 0, min = 0, sec = 0;
+
+		/* The format of date time modem info:
+		 * yy/MM/dd,hh:mm:ss±zz
+		 */
+		year = atoi(recv_buf);
+		month = atoi(recv_buf + strlen("yy/"));
+		day = atoi(recv_buf + strlen("yy/MM/"));
+		hour = atoi(recv_buf + strlen("yy/MM/dd,"));
+		min = atoi(recv_buf + strlen("yy/MM/dd,hh:"));
+		sec = atoi(recv_buf + strlen("yy/MM/dd,hh:mm:"));
+		timezone = atoi(recv_buf + strlen("yy/MM/dd,hh:mm:ss"));
+		current_date = 10000 * (2000 + year) + 100 * month + day;
+		current_time = 10000 * hour + 100 * min + sec;
+	}
+	k_delayed_work_submit(&datetime_update_work, ICAL_DATETIME_PERIOD);
+}
+
+/**@brief Callback for button events from the DK buttons and LEDs library. */
+static void button_handler(u32_t button_state, u32_t has_changed)
+{
+	u32_t button = button_state & has_changed;
+
+	if (button & UI_BUTTON_1) {
+		printk("Button presssed. Start download calendar.\n");
+		k_delayed_work_submit(&ical_work, K_NO_WAIT);
+	}
+}
+
+static int socket_sectag_set(int fd, int sec_tag)
+{
+	int err;
+	int verify;
+	sec_tag_t sec_tag_list[] = { sec_tag };
+
+	enum {
```

Comment:
I've added appropriate defines into `socket.h` some time ago, let's use them instead:
https://github.com/NordicPlayground/fw-nrfconnect-zephyr/blob/master/include/net/socket.h#L126

Reply:
Modified per the definition.